<launch>
  <arg name="simulated" default="false" />
  <group ns="HW" unless="$(arg simulated)">
    <node name="FastPWM_HAT" pkg="mavric" type="PCA9685_PWM_HAT.py" respawn="true">
      <param name="frequency" value="25000" />\
      <!-- addresses start at 0x40 (64) and can be configured on the board -->
      <param name="address" value="64" />
      <remap from="PWM_Channels" to="FastPWM" />
      <!-- we want to run as fast as possible here -->
      <!-- <param name="clk_error" value="1.0893246187363834422657952069717" /> -->
    </node>
    <node name="SlowPWM_HAT" pkg="mavric" type="PCA9685_PWM_HAT.py" respawn="true">
      <param name="frequency" value="250" />
      <!-- addresses start at 0x40 (64) and can be configured on the board -->
      <param name="address" value="65" />
      <remap from="PWM_Channels" to="SlowPWM" />
      <!-- we want to run as fast as possible here -->
      <!-- <param name="clk_error" value="1.0893246187363834422657952069717" /> -->
    </node>
	
    <node name="GPS_Streamer" pkg="mavric" type="GPS_Stream.py" respawn="true" >
      <remap from="GPS_Data" to="/GPS_Data" />
    </node>
	
    <node name="LTC2309_ADC_HAT" pkg="mavric" type="LTC2309_ADC_HAT.py" respawn="true">
        <param name="frequency" value="100" />
    </node>
  </group>
  
  <include file="$(find mavric)/launch/SystemCooling.launch" >
    <arg name="Fan_Output_Out" value="/HW/FastPWM/DutyCycleControl/CH0" />
    <arg name="Fan_Output_In" value="/HW/FastPWM/DutyCycleControl/CH1" />
    <arg name="simulated" value="$(arg simulated)" />
  </include>
  
<!--
  <group ns="Arm">
    <node name="Arm_Interface" pkg="mavric" type="Arm_P.py" respawn="true">
      <remap from="Arm_ShoulderRot" to="Command_ShoulderRot" />
      <remap from="Arm_ShoulderPitch" to="Command_ShoulderPitch" />
		
      <remap from="Arm_ElbowPitch" to="Command_ElbowPitch" />
		
      <remap from="Arm_WristRot" to="Command_WristRot" />
      <remap from="Arm_WristPitch" to="Command_WristPitch" />

      <remap from="Arm_ClawActuation" to="Command_ClawActuation" />
    </node>
  </group>

  <group ns="Drive">
    <node name="Drivetrain_Interface" pkg="mavric" type="Drive_Train_P.py" respawn="true" >
      <remap from="Drive_Train" to="Command" />
    </node>
    <node name="Drivetrain_Hartbeat" pkg="mavric" type="DrivetrainHartbeat.py" respawn="true" >
      <param name="timeout" value="1" />
      <remap from="Drive_Train" to="Command" />
    </node>
  </group>
-->

  <node name="GPS_Interface" pkg="mavric" type="GPS_Interface.py" respawn="true" />
  <node name="Temperature_Interface" pkg="mavric" type="Float64_Interface_out.py" respawn="true">
    <remap from="value" to="/SystemCooling/Temperature" />
    <param name="port" value="8002" />
  </node>
</launch>
